src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -1457,4 +1460,20 @@ Copyright (c) .NET Foundation. All rights reserved.
   -->
   <Import Project="NuGet.RestoreEx.targets" Condition="'$(RestoreUseStaticGraphEvaluation)' == 'true' And Exists('NuGet.RestoreEx.targets')" />
 
+  <!--
+    ============================================================
+    EmbedNuGetFilesInBinLog
+    Adds files to be embedded in the binary log.  These files are results of a restore
+    so they need to be added after restore is complete.
+    ============================================================
+  -->
+  <Target Name="EmbedNuGetFilesInBinLog"
+        AfterTargets="Restore"
+        Condition="'$(EmbedNuGetFilesInBinLog)' != 'false'">
+    <ItemGroup>
+      <EmbedInBinlog Include="$(ProjectAssetsFile)" Condition="Exists('$(ProjectAssetsFile)') AND $(EmbedProjectAssetsFile) != false" />
```

Comment:
I would swap the AND branches and check for EmbedProjectAssetsFile first, then check for Exists after AND. I can't remember why, but we broke someone by still hitting the disk for Exists when it was turned off.

See here:
https://github.com/dotnet/sdk/pull/20624/files

Reply:
Here was the issue where we were reading the file unnecessarily and this broke deterministic/distributed build:
https://github.com/dotnet/sdk/issues/20556